cores/rp2040/wiring_digital.cpp

```diff
@@ -90,7 +104,6 @@ extern "C" void __digitalWrite(pin_size_t ulPin, PinStatus ulVal) {
         DEBUGCORE("ERROR: Illegal pin in pinMode (%d)\n", ulPin);
         return;
     }
-    gpio_set_function(ulPin, GPIO_FUNC_SIO);
```

Comment:
While adding this in `pinMode` seems fine, I think deleting here will break the case where you have someone else controlling the pin (i.e. `tone` which uses PIO) and then you stop that process and then `digitalWrite`.  Let me whip up a simple test case...

Reply:
Yes, this breaks normal functionality.  Can you re-add the line?

Simple example blinks after analogWrite with `master`, remains PWN (connected to PIO) with this patch
````
void setup() {
  pinMode(LED_BUILTIN, OUTPUT);
  analogWriteRange(100);
  analogWrite(LED_BUILTIN, 50);
  delay(1000);
  analogWrite(LED_BUILTIN, 70);
  delay(1000);
  digitalWrite(LED_BUILTIN, LOW);
  delay(1000);
  digitalWrite(LED_BUILTIN, HIGH);
}

void loop() {
  delay(1000);
  digitalWrite(LED_BUILTIN, LOW);
  delay(1000);
  digitalWrite(LED_BUILTIN, HIGH);
}
````